torchvision/features/_bounding_box.py

```diff
@@ -0,0 +1,160 @@
+import enum
+from typing import Any, Optional, Tuple, Union
+
+import torch
+
+from ._core import TensorFeature
+
+__all__ = ["BoundingBox", "BoundingBoxFormat"]
+
+
+class BoundingBoxFormat(enum.Enum):
+    XYXY = "XYXY"
+    XYWH = "XYWH"
+    CXCYWH = "CXCYWH"
+
+
+class BoundingBox(TensorFeature):
+    formats = BoundingBoxFormat
+
+    @staticmethod
+    def _parse_format(format: Union[str, BoundingBoxFormat]) -> BoundingBoxFormat:
+        if isinstance(format, str):
+            format = format.upper()
+        return BoundingBox.formats(format)
+
+    def __init__(
+        self,
+        data: Any = None,
+        *,
+        image_size: Tuple[int, int],
+        format: Union[str, BoundingBoxFormat],
+    ):
+        super().__init__()
+        self._image_size = image_size
+        self._format = self._parse_format(format)
+
+        self._convert_to_xyxy = {
+            self.formats.XYWH: self._xywh_to_xyxy,
+            self.formats.CXCYWH: self._cxcywh_to_xyxy,
+        }
+        self._convert_from_xyxy = {
+            self.formats.XYWH: self._xyxy_to_xywh,
+            self.formats.CXCYWH: self._xyxy_to_cxcywh,
+        }
+
```

Comment:
For when we will be discussing specifics of the data classes, it might be good to add a `__torch_function__` that forbids applying operations of two bounding boxes if they have different `image_size`, while allowing `BoundingBox` + `Tensor` to work